packages/fluentui/docs/src/prototypes/VirtualizedStickyTree/VirtualizedStickyTree.tsx

```diff
@@ -0,0 +1,465 @@
+          onFocusFirstChild: data.handleFocusFirstChild,
+        });
+      })}
+      {children}
+      {stickyBottomItemsIndex.map((itemIndex: number, index: number, arr: number[]) => {
+        const item = renderedItems[itemIndex];
+        return React.cloneElement(item, {
+          style: {
+            position: 'sticky',
+            height: stickyItemSize,
+            zIndex: 111,
+            top: height - (arr.length - index) * stickyItemSize,
+          },
+          onTitleClick: onClickSticky,
+          onFocusParent: data.handleFocusParent,
+          onFocusFirstChild: data.handleFocusFirstChild,
+        });
+      })}
+    </div>
+  );
+};
+
+const calcRenderedHeight = (renderedItems: React.ReactElement[], stickyItemSize: number, itemSize: number) => {
+  let renderedHeight = 0;
+  renderedItems.forEach(item => {
+    renderedHeight = renderedHeight + (item.props.level === 1 ? stickyItemSize : itemSize);
+  });
+  return renderedHeight;
+};
+
+interface ReRenderStickyTreeProps {
+  renderedItems: React.ReactElement<TreeItemProps>[];
+  stickyItemSize: number;
+  itemSize: number;
+  height: number;
+  toplvlItemsId: string[];
+  onActiveItemsIdChange: (idsToRemove?: string[], idsToAdd?: string[]) => void;
+}
+
+interface ItemDataType {
+  renderedItems: React.ReactElement<TreeItemProps>[];
+  isNonSticky: (item: React.ReactElement<TreeItemProps>) => boolean;
+  onClick: (event: React.SyntheticEvent<HTMLElement>, itemProps: TreeItemProps) => void;
+  handleOnFocus: (e: React.SyntheticEvent<HTMLElement>) => void;
+  handleFocusParent: (event: React.SyntheticEvent<HTMLElement>, treeItemProps: TreeItemProps) => void;
+  handleFocusFirstChild: (event: React.SyntheticEvent<HTMLElement>, treeItemProps: TreeItemProps) => void;
+  handleOnSiblingsExpand: (event: React.SyntheticEvent<HTMLElement>, treeItemProps: TreeItemProps) => void;
+}
+
+const ReRenderStickyTree = ({
+  renderedItems,
+  stickyItemSize,
+  itemSize,
+  height,
+  toplvlItemsId,
+  onActiveItemsIdChange,
+}: ReRenderStickyTreeProps) => {
+  const [stickyTopIds, setStickyTopIds] = React.useState<string[]>([toplvlItemsId[0]]);
+  const [stickyBottomIds, setStickyBottomIds] = React.useState<string[]>(toplvlItemsId.slice(1));
+
+  const passOnStickyTopIds = React.useCallback((newStickyTopIds: string[]) => {
+    setStickyTopIds([...newStickyTopIds]);
+  }, []);
+  const passOnStickyBottomIds = React.useCallback((newStickyBottomIds: string[]) => {
+    setStickyBottomIds([...newStickyBottomIds]);
+  }, []);
+
+  // when navigate with keyboard, use `itemIdTobeFocused` to remember the focused item id,
+  // then when DOM updates are finished, sticky tree knows which item to focus on
+  const [itemIdTobeFocused, setItemIdTobeFocused] = React.useState<string>(null);
+  const passItemIdTobeFocused = React.useCallback((newItemIdTobeFocused: string) => {
+    setItemIdTobeFocused(newItemIdTobeFocused);
+  }, []);
+
+  React.useLayoutEffect(() => {
```

Comment:
You might not need `useLayoutEffect` here, since you're only scrolling to a specific item so you don't need to run this logic before the browser paint.... a simply `useEffect` here should do

Reply:
Good point. Thank you for the explanation. I tested with useEffect and it works well. Going to make a commit